build.py

```diff
@@ -5,245 +5,394 @@
 
 import argparse
 import os
+import platform
 import shutil
-import subprocess
 import sys
 import warnings
 
+from pathlib import Path
 
-def is_windows():
-    """Check if the current platform is Windows."""
-    return sys.platform.startswith("win")
+REPO_ROOT = Path(__file__).parent
+sys.path.append(str(REPO_ROOT / "tools" / "python"))
```

Comment:
```suggestion
sys.path.append(os.fspath(REPO_ROOT / "tools" / "python"))
```

Reply:
Does the pathlib.Path str() implementation not handle this?

Reply:
pathlib.Path str() will handle this. This can become a problem when the type of input is not known to be of type pathlib.Path. For example, if REPO_ROOT was not a pathlib.Path type, it would lead to unexpected errors.

In this scenario, I think it doesn't matter. So, I think it should be ok to leave this as is.